hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/DefaultNearCacheManager.java

```diff
@@ -103,6 +93,16 @@ public DefaultNearCacheManager(SerializationService ss, TaskScheduler es,
                 scheduler, classLoader, properties);
     }
 
+    @Override
+    public void startPreloading(NearCache nearCache, DataStructureAdapter dataStructureAdapter) {
+        NearCacheConfig nearCacheConfig = nearCache.getNearCacheConfig();
+        NearCachePreloaderConfig preloaderConfig = nearCacheConfig.getPreloaderConfig();
+        if (preloaderConfig.isEnabled()) {
+            createAndSchedulePreloadTask(nearCache, dataStructureAdapter);
```

Comment:
Is it safe to schedule the preloader without owning the mutex? What would be the effect of concurrently destroying the near cache (eg in create-use-destroy style of test)?

Reply:
Yes it should be safe. Goal is only schedule it 1 time. Now we provide it by using atomic-ref's cas [here](https://github.com/hazelcast/hazelcast/pull/16310/files#diff-7815fb4175dee56daeb7a8588a3d94c9R170) when creating. Concurrent destroys should also be safe.